src/drivers/fluid_winmidi.c

```diff
@@ -145,27 +199,101 @@ fluid_winmidi_callback(HMIDIIN hmi, UINT wMsg, DWORD_PTR dwInstance,
+ * devices index (see fluid_winmidi_midi_driver_settings()).
+ *
+ * @param dev_idx, device index.
+ * @param dev_name, name of the device.
+ * @return the new device name (that must be freed when finish with it) or
+ *  NULL if memory allocation error.
+ */
+static char *fluid_winmidi_get_device_name(int dev_idx, char *dev_name)
+{
+    char *new_dev_name;
+
+    int i =  dev_idx;
+    size_t size = 0; /* index size */
+
+    do
+    {
+        size++;
+        i = i / 10 ;
+    }
+    while(i);
+
+    /* index size + separator + name length + zero termination */
+    new_dev_name = FLUID_MALLOC(size + 2 + FLUID_STRLEN(dev_name));
+    if(new_dev_name)
+    {
+        /* the name is filled if allocation is successful */
+        FLUID_SPRINTF(new_dev_name, "%d:%s", dev_idx, dev_name);
+    }
+    else
+    {
+        FLUID_LOG(FLUID_ERR, "Out of memory");
+    }
+    return new_dev_name;
+}
+
+/*
+  Internal multi device name template.
+*/
+const static char *multi_dev_name = "x[;y;z;..]";
+
+/*
+ Add setting midi.winmidi.device in the settings.
+
+ MIDI devices names are enumerated and added to midi.winmidi.device setting
+ options. Example:
+ 0:Port MIDI SB Live! [CE00], 1:SB PCI External MIDI, default, x[;y;z;..]
+
+ Devices name prefixed by index (i.e 1:SB PCI External MIDI) are real devices.
+ "default" name is the default device.
+ "x[;y;z;..]" is the multi device naming. Its purpose is to indicate to
+ the user how he must specify a multi device name in the setting.
+ A multi devices name must be a list of real devices index separated by semicolon:
+ Example: "5;3;0"
+*/
 void fluid_winmidi_midi_driver_settings(fluid_settings_t *settings)
 {
     MMRESULT res;
     MIDIINCAPS in_caps;
     UINT i, num;
+
+    /* register midi.winmidi.device */
     fluid_settings_register_str(settings, "midi.winmidi.device", "default", 0);
     num = midiInGetNumDevs();
 
     if(num > 0)
     {
         fluid_settings_add_option(settings, "midi.winmidi.device", "default");
 
+        /* add real devices names in options list */
         for(i = 0; i < num; i++)
         {
             res = midiInGetDevCaps(i, &in_caps, sizeof(MIDIINCAPS));
 
             if(res == MMSYSERR_NOERROR)
             {
-                fluid_settings_add_option(settings, "midi.winmidi.device", in_caps.szPname);
+                /* add new device name (prefixed by its index) */
+                char *new_dev_name = fluid_winmidi_get_device_name(i, in_caps.szPname);
+                if(!new_dev_name)
+                {
+                    break;
+                }
+                fluid_settings_add_option(settings, "midi.winmidi.device",
+                                          new_dev_name);
+                FLUID_FREE(new_dev_name);
             }
         }
+
+        /* add multi device name template among other real devices names */
+        fluid_settings_add_option(settings, "midi.winmidi.device", multi_dev_name);
```

Comment:
I think it's confusing to add this device name template as option to the setting. Instead, this template should be documented in fluidsettings.xml.